### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by yusufobr**
Hello dear reviewer, for this milestone, I :

-  Initialised React application using create-react-app
-  Fetched data from CoinApp API 
-  Filtered stored data by their rank
-  Displayed the data along with the crypto name
-  When the user clicks on the crypto name, the application navigates to the details page.
-  Retrieved data from the API to show detailed data in the category.
-  When the user clicks on the "Back" button, the user navigates to the home page.
-  Wrote unit tests for pure functions in redux code.
-  Wrote integration tests using the React Testing Library.

[Video demo](https://www.loom.com/share/0430a9cf864e4396b0f8729094bf0a6f)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,167 @@
...
-This section has moved here: [https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size](https://facebook.github.io/create-react-app/docs/analyzing-the-bundle-size)
+### Setup
 
-### Making a Progressive Web App
+Clone this repository to your desired folder:
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app](https://facebook.github.io/create-react-app/docs/making-a-progressive-web-app)
+```sh
+  cd my-folder
+  git clone https://github.com/yusufobr/crypto-coins.git
```
Comment (descholar-ceo) [inline README.md]: - [ ] This link is not correct, your repo's link should be: `https://github.com/yusufobr/crypto-data.git` so please fix your instruction as follows:
```suggestion
  git clone https://github.com/yusufobr/crypto-data.git
```

File: README.md [hunk 2]
```diff
@@ -1,70 +1,167 @@
...
 
-### Advanced Configuration
+### Install
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+Install this project with:
 
-### Deployment
+```sh
+  cd crypto-coins
```
Comment (descholar-ceo) [inline README.md]: - [ ] After cloning with fixed kink at the top, they will `cd crypto-data`
```suggestion
  cd crypto-data
```

File: README.md [hunk 3]
```diff
@@ -1,70 +1,167 @@
...
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+### Run tests
+
+To run tests, run the following command:
+
+```sh
+  npx hint .
+  npx stylelint "\*_/_.{css,scss}"
+  npx eslint "**/*.{js,jsx}"
+```
```
Comment (descholar-ceo) [inline README.md]: - [ ] The commands you specified are not for running test cases, these commands are for running linters instead, for running test cases they will have to execute this command: `npm run test` <= Would you fix the instruction accordingly?

File: src/__tests__/App.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import App from '../App';
+
+describe('App.js', () => {
+  it('should render correctly', () => {
+    const tree = renderer.create(
+      <Provider store={store}>
+        <App />
+      </Provider>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (descholar-ceo) [inline src/__tests__/App.test.js:13]: - [ ] This test case is failing, with the error pasted in the screenshot below
Screenshot 2023-06-01 at 17 02 20

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @yusufobr 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 You have done an amazing project, this is one of the best projects I have ever seen there, keep up the good job, this looks great 💪🏽 🍾 🎉 🎈 ✔️ 
👍🏽 Readme file looks very professional, with attention to detail, except just the very small touches that it needs ✔️ 
👍🏽 PR description and commit messages are so professional ✔️ 
👍🏽 The PR passes all linters checks ✔️ 
👍🏽 You are following the correct git flow ✔️ 
👍🏽 The project is deployed and the video for the demo is recorded ✔️ 

## Required Changes ♻️
_Check the comments under the review._
However there are some small room for improvement, as you may know, we always have somewhere to improve, that's we grow from others, so I will list down those small touches that you will make, in order to move on to the next step:
- [ ] The recorded video, doesn't show the project's presentation, you will have to re-record again and make sure that you are sharing your screen correctly so that we will be able to see your project while presenting, currently, we only see Loom's page and hear your voice explaining the project but without seeing it live in the video, kindly note that the video has to highlight the following points as stated in the project's description:
     - [ ] The video should be presenting the features of the project.
     - [ ] The maximum length of the video is 5 minutes.
     - [ ] The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
     - [ ] The video contains a demo of the project, highlighting all the project features.
     - [ ] The video contains a highlight about a piece of the code the student finds interesting or is proud of making.
- [ ] While writing test cases, you forgot to test the following components:
        - [ ] `Chart.js`
        - [ ] `Coin.js`
- [ ] You should also add test cases for the following helper functions:
        - [ ] `adjustNumber` from the file `src/helpers/adjustNumber.js`
- [ ] And also, as the project's requirements say, you will need to perform test cases for `some pure functions in Redux code`
- [ ] Last but not least, you perform a mocking data fetching and Redux Store connection.

## Optional suggestions
- [ ] I would recommend that next time when you submit a project which has test cases, you try to install them on Github actions so that you are sure that they pass every time you make some changes, by looking at the GitHub actions, it makes it so easy for you to detect even small failures that you get ⚠️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/App.test.js (hunk 4 above)
Comment (yusufobr) [reply src/__tests__/App.test.js:13]: done

**Comment by yusufobr** [comment]
Hi @descholar-ceo, thank you for the review it was well-detailed.  However:
- In regard to testing the `Chart`, I'm unable to test the component because it's based on an external library. 

_I would appreciate some feedback on how to go about this required change._ 💯

**Comment by descholar-ceo** [comment]
> Hi @descholar-ceo, thank you for the review it was well-detailed. However:
> 
> * In regard to testing the `Chart`, I'm unable to test the component because it's based on an external library.
> 
> _I would appreciate some feedback on how to go about this required change._ 💯

Just for your learning purposes, you would try out this question on StackOverflow: [Writing unit tests with react testing library for recharts](https://stackoverflow.com/questions/73117667/writing-unit-tests-with-react-testing-library-for-recharts). While I am not sure if you tried testing it and might have failed due to any reason, but from what I saw that most of the people get as a challenge when they write test cases for their components depending on `rechart` package, is something related to responsiveness rendering, you would check out this issue on the `rechart's` official repo: [ResponsiveContainer cannot be rendered and hence cannot be tested using react testing library #2268
](https://github.com/recharts/recharts/issues/2268) 

Just be bold, and try to push your limit, I am sure that your effort will be beneficial to you very soon, whenever you get a blocker or any issue feel free to post it here on the thread!

File: README.md (hunk 2 above)
Comment (yusufobr) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (yusufobr) [reply README.md]: done

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @yusufobr  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks are passing :heavy_check_mark: 
- All the tests are passing :heavy_check_mark: 
- The video you recorded met all requirements :surfer: 
- Your website looks really good :star_struck: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**